src/libsync/utility_win.cpp

```diff
@@ -15,21 +15,36 @@
 #include <shlobj.h>
 #include <winbase.h>
 #include <windows.h>
+#include <shlguid.h>
+#include <string>
+#include <QLibrary>
 
 static const char runPathC[] = "HKEY_CURRENT_USER\\Software\\Microsoft\\Windows\\CurrentVersion\\Run";
 
+
+     
 static void setupFavLink_private(const QString &folder)
 {
     // Windows Explorer: Place under "Favorites" (Links)
-    wchar_t path[MAX_PATH];
-    SHGetSpecialFolderPath(0, path, CSIDL_PROFILE, FALSE);
-    QString profile =  QDir::fromNativeSeparators(QString::fromWCharArray(path));
+    
+    QString linkName; 
     QDir folderDir(QDir::fromNativeSeparators(folder));
-    QString linkName = profile+QLatin1String("/Links/") + folderDir.dirName() + QLatin1String(".lnk");
+     
+    /* Use new WINAPI functions */
+    PWSTR path;
+    
+    if(SHGetKnownFolderPath(FOLDERID_Links, 0, NULL, &path) == S_OK) {
+        QString links = QDir::fromNativeSeparators(QString::fromWCharArray(path)); 
+        linkName = QDir(links).filePath(folderDir.dirName() + QLatin1String(".lnk"));
+    }
+    qDebug() << Q_FUNC_INFO << " creating link from " << linkName << " to " << folder;
+    CoTaskMemFree(path);
```

Comment:
This will crash if path is not null and yet not pointing to a valid allocation, which can happen if `SHGetKnownFolderPath` doesn't touch it. You could either move CoTaskMemFree into the `if(){}`, or initialize `path` to `NULL`.